jwst/ramp_fitting/ramp_fit.py

```diff
@@ -105,9 +118,286 @@ def ramp_fit(model, buffsize, save_opt, readnoise_model, gain_model,
+           Object containing optional GLS-specific ramp fitting data for the
+           exposure
+       """
+    if max_cores == 'none':
+        number_slices = 1
+    else:
+        num_cores = multiprocessing.cpu_count()
+        log.debug("Found %d possible cores to use for ramp fitting " % num_cores)
+        if max_cores == 'quarter':
+            number_slices = num_cores // 4 or 1
+        elif max_cores == 'half':
+            number_slices = num_cores // 2 or 1
+        elif max_cores == 'all':
+            number_slices = num_cores
+        else:
+            number_slices = 1
+    if pipe_utils.is_tso(input_model) and hasattr(input_model, 'int_times'):
+        int_times = input_model.int_times
+    else:
+        int_times = None
+    total_rows = input_model.data.shape[2]
+    total_cols = input_model.data.shape[3]
+    number_of_integrations = input_model.data.shape[0]
+    if number_slices == 1:
+        max_segments, max_CRs = calc_num_seg(input_model.groupdq, number_of_integrations)
+        int_model, opt_model, out_model = create_output_models(input_model,
+                                            number_of_integrations, save_opt, total_cols, total_rows,
+                                                               max_segments, max_CRs)
+
+        out_model.data, out_model.dq, out_model.var_poisson, out_model.var_rnoise, out_model.err,\
+        int_data, int_dq, int_var_poisson, int_var_rnoise, int_err,\
+        dummy, opt_slope, opt_sigslope, opt_var_poisson, opt_var_rnoise, \
+        opt_yint, opt_sigyint, opt_pedestal, opt_weights, opt_crmag,\
+        actual_segments, actual_CRs = \
+        ols_ramp_fit(input_model.data, input_model.err, input_model.groupdq, input_model.pixeldq,
+                     buffsize, save_opt, readnoise_2d, gain_2d, weighting,
+                          input_model.meta.instrument.name, input_model.meta.exposure.frame_time,
+                          input_model.meta.exposure.ngroups, input_model.meta.exposure.group_time,
+                          input_model.meta.exposure.groupgap, input_model.meta.exposure.nframes,
+                          input_model.meta.exposure.drop_frames1, int_times)
+        if number_of_integrations > 1:
+            int_model.data = int_data
+            int_model.dq = int_dq
+            int_model.var_poisson = int_var_poisson
+            int_model.var_rnoise = int_var_rnoise
+        if save_opt:
+            opt_model.slope = opt_slope
+            opt_model.sigslope = opt_sigslope
+            opt_model.var_poisson = opt_var_poisson
+            opt_model.var_rnoise = opt_var_rnoise
+            opt_model.yint = opt_yint
+            opt_model.sigyint = opt_sigyint
+            opt_model.pedestal = opt_pedestal
+            opt_model.weights = opt_weights
+            opt_model.crmag = opt_crmag
+        return out_model, int_model, opt_model
+    else:
+        log.debug("number of processes being used is %d" % number_slices)
```

Comment:
update to `log.debug(f'number of processes being used is {number_slices}')`

Reply:
changed